test/AlgebraicGeometry/Schemes/DerivedPushforward.jl

```diff
@@ -0,0 +1,272 @@
+  @test iszero(H2)
+
+  B3, _ = image(map(st0, 1, (-2,)))
+  H3, _ = quo(st0[-3], B3)
+  @test !iszero(H3)
+
+  # Compute the higher direct images of the structure sheaf
+  S1 = graded_free_module(S, [0])
+  I = ideal(S, f)
+  IS1, inc = I*S1
+  M = cokernel(inc)
+  st0 = Oscar._derived_pushforward(M);
+
+  Z0, _ = kernel(map(st0, 1, (0,)))
+  @test !iszero(Z0)
+  @test is_projective(Z0)[1] # H^{0, 0} does not change throughout the family
+
+  B1, _ = image(map(st0, 1, (0,)))
+  Z1, _ = kernel(map(st0, 1, (-1,)))
+
+  H1, _ = quo(Z1, B1)
+  @test iszero(H1) # H^{1, 0} = 0 for K3-surfaces
+
+  B2, _ = image(map(st0, 1, (-1,)))
+  Z2, _ = kernel(map(st0, 1, (-2,)))
+
+  H2, _ = quo(Z2, B2)
+  @test !iszero(H2)
+  @test is_projective(H2)[1] # H^{2, 0} is constant in this family
+
+  B3, _ = image(map(st0, 1, (-2,)))
+  H3, _ = quo(st0[-3], B3)
+  @test iszero(H3)
+  
+
+  M1 = Oscar.relative_cotangent_module(IPX)
+  M1, _ = pushforward(inc_X, M1)
+  st = Oscar._derived_pushforward(M1);
+
+  stsp = simplify(Oscar.SimpleComplexWrapper(st));
+  A = matrix(map(stsp, 1, (0,)))
+  B = matrix(map(stsp, 1, (-1,)))
+
+  # Test for a smooth fiber: H¹(X₁, Ω¹) = ℚ ²⁰.
+  A1 = map_entries(f->f(one(ZZ)), A)
+  B1 = map_entries(f->f(one(ZZ)), B)
+  @test rank(stsp[-1]) - rank(A1) - rank(B1) == 20
+
+  # Test for the singular fiber: H¹(X₀, Ω¹) = ℚ ²³.
+  A0 = map_entries(f->f(zero(ZZ)), A)
+  B0 = map_entries(f->f(zero(ZZ)), B)
+  @test rank(stsp[-1]) - rank(A0) - rank(B0) == 23
+end
+
+@testset "K3 surface over a field" begin
+  A = QQ
+  P = projective_space(A, [:x, :y, :z, :w])
+  S = homogeneous_coordinate_ring(P)
+  n = ngens(S)
+  f1 = sum(x^n for x in gens(S))
+  f0 = prod(gens(S))
+  f = f1 + 2*f0
+
+  IPX, inc_X = sub(P, f)
+  Sm4 = graded_free_module(S, [4])
+
+  # Compute the higher direct images of the structure sheaf
+  S1 = graded_free_module(S, [0])
+  I = ideal(S, f)
+  IS1, inc = I*S1
+  M = cokernel(inc)
+  st0 = Oscar._derived_pushforward(M);
+  st0_simp = simplify(st0);
+  
+  @test rank(st0_simp[0]) == 1
+  @test iszero(st0_simp[-1])
+  @test rank(st0_simp[-2]) == 1
+  @test iszero(st0_simp[-3])
+
+  # Compute the higher direct images of \Omega^1
+  M1 = Oscar.relative_cotangent_module(IPX)
+  M1, _ = pushforward(inc_X, M1)
+  st = Oscar._derived_pushforward(M1);
+  stsp = simplify(st);
+
+  @test iszero(stsp[0])
+  @test rank(stsp[-1]) == 20
+  @test iszero(stsp[-2])
+end
+
+@testset "mild log singularities" begin
+  B, (t,) = polynomial_ring(QQ, [:t])
+
+  IP5 = projective_space(B, [:u, :v, :w, :x, :y, :z])
+  S = homogeneous_coordinate_ring(IP5)
+  (u, v, w, x, y, z) = gens(S)
+
+  f = x*y - t^2*v^2 - w^2
+  g = z*w - t*u*v
+
+  I = ideal(S, [f, g])
+
+  X, inc_X = sub(IP5, I)
+
+  # Testing the structure sheaf
+  M0, _ = pushforward(inc_X, graded_free_module(homogeneous_coordinate_ring(X), [0]))
+
+  C = simplify(Oscar._derived_pushforward(M0));
+
+  @test rank(C[0]) == 1 # Free of rank one
+  @test all(k->iszero(C[-k]), 1:5)
+
```

Comment:
Here we now also test the structure sheaf.

(Comment made for visibility to certain externals)